pkg/registry/daemon/rest.go

```diff
@@ -0,0 +1,124 @@
+	"k8s.io/kubernetes/pkg/api/validation"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/registry/generic"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/fielderrors"
+)
+
+// daemonStrategy implements verification logic for daemons.
+type daemonStrategy struct {
+	runtime.ObjectTyper
+	api.NameGenerator
+}
+
+// Strategy is the default logic that applies when creating and updating Daemon objects.
+var Strategy = daemonStrategy{api.Scheme, api.SimpleNameGenerator}
+
+// NamespaceScoped returns true because all Daemons need to be within a namespace.
+func (daemonStrategy) NamespaceScoped() bool {
+	return true
+}
+
+// PrepareForCreate clears the status of a daemon before creation.
+func (daemonStrategy) PrepareForCreate(obj runtime.Object) {
+	daemon := obj.(*api.Daemon)
+	daemon.Status = api.DaemonStatus{}
+
+	daemon.Generation = 1
+}
+
+// PrepareForUpdate clears fields that are not allowed to be set by end users on update.
+func (daemonStrategy) PrepareForUpdate(obj, old runtime.Object) {
+	newDaemon := obj.(*api.Daemon)
+	oldDaemon := old.(*api.Daemon)
+
+	// Any changes to the spec increment the generation number, any changes to the
+	// status should reflect the generation number of the corresponding object. We push
+	// the burden of managing the status onto the clients because we can't (in general)
+	// know here what version of spec the writer of the status has seen. It may seem like
+	// we can at first -- since obj contains spec -- but in the future we will probably make
+	// status its own object, and even if we don't, writes may be the result of a
+	// read-update-write loop, so the contents of spec may not actually be the spec that
+	// the controller has *seen*.
+	//
+	// TODO: Any changes to a part of the object that represents desired state (labels,
+	// annotations etc) should also increment the generation.
+	if !reflect.DeepEqual(oldDaemon.Spec, newDaemon.Spec) {
+		newDaemon.Generation = oldDaemon.Generation + 1
+	}
+}
+
+// Validate validates a new daemon.
+func (daemonStrategy) Validate(ctx api.Context, obj runtime.Object) fielderrors.ValidationErrorList {
+	daemon := obj.(*api.Daemon)
+	return validation.ValidateDaemon(daemon)
+}
+
+// AllowCreateOnUpdate is false for daemon; this means a POST is
+// needed to create one
+func (daemonStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+// ValidateUpdate is the default update validation for an end user.
+func (daemonStrategy) ValidateUpdate(ctx api.Context, obj, old runtime.Object) fielderrors.ValidationErrorList {
+	validationErrorList := validation.ValidateDaemon(obj.(*api.Daemon))
+	updateErrorList := validation.ValidateDaemonUpdate(old.(*api.Daemon), obj.(*api.Daemon))
+	return append(validationErrorList, updateErrorList...)
+}
+
+func (daemonStrategy) AllowUnconditionalUpdate() bool {
+	return true
+}
+
+// DaemonToSelectableFields returns a label set that represents the object.
```

Comment:
Would it be slightly more correct to say it returns a field set instead of a label set?